packages/server/src/utils/ldap.ts

```diff
@@ -0,0 +1,74 @@
+import { Client } from 'ldapts';
+import { User } from '../services/database/types';
+import Logger from '@joplin/utils/Logger';
+import { LdapConfig } from './types';
+
+export default async function ldapLogin(email: string, password: string, user: User, config: LdapConfig): Promise<User> {
+	const logger = Logger.create('LDAP');
+
+	const enabled = config.enabled;
+	const userCreation = config.userCreation;
+	const host = config.host;
+	const mailAttribute = config.mailAttribute;
+	const fullNameAttribute = config.fullNameAttribute;
+	const baseDN = config.baseDN;
+	const bindDN = config.bindDN;
+	const bindPW = config.bindPW;
+
+	logger.info(`Starting authentication with Server ${host}`);
+
+	if (password === '') {
+		logger.error('no password entered');
+		throw new Error('no password entered');
+	}
+
+	if (enabled) {
+		let searchResults;
+		const client = new Client({
+			url: host,
+			timeout: 2000,
+			connectTimeout: 1000,
+		});
+
+		if (bindDN.length !== 0) {
+			try {
+				await client.bind(bindDN, bindPW);
+			} catch (ex) {
+				throw new Error('Could not bind to LDAP server.');
```

Comment:
To keep the stack trace you should reuse the same error. It should be like this:

```
error.message = "Could not bind the LDAP server: " + error.message;
throw error
```

(And please name it "error" for consistency with the rest of the code)